src/homework/sandro/FourInARow/FourInARow.java

```diff
@@ -0,0 +1,305 @@
+package homework.sandro.FourInARow;
+
+import homework.sandro.Colors;
+
+import java.util.Scanner;
+
+public class FourInARow {
+//    static {
+//        if (Math.random() > 0.5) {
+//            throw new RuntimeException("Hi Sandro, was geht ab?");
+//        }
+//    }
+
+    public static void main(String[] args) {
+
+        int[][] board = new int[6][7];
+        int currentplayer = 1;
+        boolean winner = false;
+        boolean empty = true;
+
+        printBoard(board, 6, 7);
+        while (!winner && empty) {                       //Game runs
+            int col;
+            int row;
+            while (true) {                               //Input read + check if legal move and column empty proof
+                col = readInt("pick a Col, Player Number: " + currentplayer);
+                row = emptyColumn(board, col);
+                if (row == -1) {
```

Comment:
I suggest that you reconsider your approach in checking if a game is won. Right now, you always have to go over the entire field with loops and such. Here you actually have the col and row of the LATEST move. Instead of now checking the entire board, check your board against the latest move. 

You can still use your loops, but make them more efficient. Due to the latest move col, you only need to check one column for vertical win, not all. Due to the latest row, you only need to check one row for win, not all.

With this, checking diagonally is also easier. What do you need to check based on your latest move for diagonal stuff? If you check from your stone 3 positions top left and 3 positions bottom right, you have all the values which could lead to a win from your current position topLeftToBottom right [row-3][col-3], [row-2][col-2], [row-1][col-1], [row+1][col+1], [row+2][col+2], [row+3][col+3]. The same can be done with a slight modification for bottomLeftToTopRight.

You can do this in loops, the concept of checking diagonals based on the latest move is what you see in my recursion. But you do not need recursion, do it with loops and figure out how it works for you.

But the main takeaway is.. Don't check the entire board. Just check the latest move. Safes you work


I will look into the other stuff at a later time, gotta go now :D But if you want to try another approach in the meantime, feel free